systemtest/src/test/java/io/strimzi/systemtest/operators/topic/ThrottlingQuotaST.java

```diff
@@ -0,0 +1,359 @@
+import io.strimzi.systemtest.AbstractST;
+import io.strimzi.systemtest.Constants;
+import io.strimzi.systemtest.annotations.ParallelTest;
+import io.strimzi.systemtest.resources.crd.kafkaclients.KafkaAdminClient;
+import io.strimzi.systemtest.resources.crd.kafkaclients.AdminClientOperations;
+import io.strimzi.systemtest.resources.operator.SetupClusterOperator;
+import io.strimzi.systemtest.templates.crd.KafkaTemplates;
+import io.strimzi.systemtest.templates.crd.KafkaUserTemplates;
+import io.strimzi.systemtest.utils.ClientUtils;
+import io.strimzi.systemtest.utils.kafkaUtils.KafkaTopicUtils;
+import io.strimzi.systemtest.utils.kafkaUtils.KafkaUtils;
+import io.strimzi.systemtest.utils.kubeUtils.controllers.JobUtils;
+import io.strimzi.systemtest.utils.kubeUtils.objects.PodUtils;
+import org.apache.kafka.common.security.auth.SecurityProtocol;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+import static io.strimzi.systemtest.Constants.GLOBAL_TIMEOUT;
+import static io.strimzi.systemtest.Constants.INTERNAL_CLIENTS_USED;
+import static io.strimzi.systemtest.Constants.REGRESSION;
+import static io.strimzi.test.k8s.KubeClusterResource.kubeClient;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+
+@Tag(REGRESSION)
+@Tag(INTERNAL_CLIENTS_USED)
+public class ThrottlingQuotaST extends AbstractST {
+
+    private static final Logger LOGGER = LogManager.getLogger(ThrottlingQuotaST.class);
+    static final String NAMESPACE = "throttling-quota-cluster-test";
+    static final String CLUSTER_NAME = "quota-cluster";
+    final String classTopicPrefix = "quota-topic-test";
+
+    /**
+     * Test checks for throttling quotas set for user
+     * on creation & deletion of topics and create partition operations.
+     */
+    @ParallelTest
+    void testThrottlingQuotasCreateTopic(ExtensionContext extensionContext) {
+        final String kafkaUsername = mapWithTestUsers.get(extensionContext.getDisplayName());
+        final String createAdminName = "create-admin-" + mapWithKafkaClientNames.get(extensionContext.getDisplayName());
+        final String topicNamePrefix = classTopicPrefix + "-create";
+        int topicsCountOverQuota = 500;
+        setupKafkaUserInNamespace(extensionContext, kafkaUsername);
+
+        KafkaAdminClient adminClientJob = new KafkaAdminClient.Builder()
+                .withAdminName(createAdminName)
+                .withBootstrapAddress(KafkaResources.plainBootstrapAddress(CLUSTER_NAME))
+                .withTopicName(topicNamePrefix)
+                .withTopicCount(topicsCountOverQuota)
```

Comment:
yeah, it made sense to me. Don't kafka CLI tools work the same way, when you want to create multiple topics at once?

Reply:
The Kafka CLI tools don't support creating multiple topics at once. IMHO if you really need a type safe API for this it would be a cleaner to just support a collection of topic names to be created. Or _possibly_ use a little helper class with a prefix and number of topics to represent multiple topic names.